packages/jest-jasmine2/src/jasmine/Spec.js

```diff
@@ -148,12 +146,12 @@ Spec.prototype.onException = function onException(error) {
     return;
   }
 
-  if (error instanceof ExpectationFailed) {
-    return;
+  if (error && error.code === 'ERR_ASSERTION') {
```

Comment:
not happy with this. The assertion itself is fine (probably better than `instanceof`), but this shouldn't be a necessary change